### PR TITLE
parse args with argparse Fixes #9

### LINKED ISSUE #9: support `--ignore=x,y,z` ignore in `args: [`
re-opening https://github.com/Lucas-C/pre-commit-hooks-safety/issues/7

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

**Comment by graingert** [comment]
this supports nicer `--ignore=1,2,3` syntax and allows requirements files starting with `-` to be passed to safety.check.main

**Comment by graingert** [comment]
@Lucas-C it looks like travis is disconnected from github

**Comment by Lucas-C** [comment]
Thanks for the contribution !

FYI, I rebased your branch on `master`.

Sadly, the Travis CI checks are not passing:
https://travis-ci.org/github/Lucas-C/pre-commit-hooks-safety

Why adding a `--full-report` option ? I'm not sure it's needed, I think `--short-report` is enough.
Also, could you please add a test for this new flag ?

File: pre_commit_hooks/safety_check.py [hunk 1]
```diff
@@ -1,18 +1,38 @@
...
+        "--full-report",
+        default="--full-report",
+        const="--full-report",
+        action="store_const",
+    )
+    parser.add_argument(
+        "--short-report", dest="full_report", const="--short-report", action="store_const"
+    )
+    parser.add_argument("--ignore", action="append")
+    parser.add_argument("files", nargs="*", action="append")
```
Comment (graingert) [inline pre_commit_hooks/safety_check.py:21]:
```suggestion
    parser.add_argument("files", nargs="*")
```

File: pre_commit_hooks/safety_check.py [hunk 2]
```diff
@@ -1,18 +1,38 @@
...
+    ns, rest = parser.parse_known_args(argv)
+    ignore = [
+        arg
+        for vs in ns.ignore
```
Comment (graingert) [inline pre_commit_hooks/safety_check.py:25]:
```suggestion
        for vs in ns.ignore or []
```

**Comment by graingert** [comment]
> Why adding a --full-report option ? I'm not sure it's needed, I think --short-report is enough.

this allows `--full-report` to be enabled by default

File: pre_commit_hooks/safety_check.py [hunk 3]
```diff
@@ -1,18 +1,38 @@
...
+    parser.add_argument(
+        "--full-report",
+        default="--full-report",
+        const="--full-report",
+        action="store_const",
+    )
+    parser.add_argument(
+        "--short-report", dest="full_report", const="--short-report", action="store_const"
+    )
+    parser.add_argument("--ignore", action="append")
```
Comment (Lucas-C) [inline pre_commit_hooks/safety_check.py:20]: Why specifying this ? Shouldn't it be passed through to `safety_check` in the `rest` variable ?
I do not want to re-define all existing / supported `safety` flags here

**Comment by Lucas-C** [comment]
> this allows --full-report to be enabled by default
I think it should be doable without defining an extra option :)

File: pre_commit_hooks/safety_check.py (hunk 3 above)
Comment (graingert) [reply pre_commit_hooks/safety_check.py]: this is because I want to support `--ignore=1,2,3`

**Comment by graingert** [comment]
> > this allows --full-report to be enabled by default
> 
> I think it should be doable without defining an extra option :)

it only defines one option: full_report

File: pre_commit_hooks/safety_check.py (hunk 3 above)
Comment (graingert) [reply pre_commit_hooks/safety_check.py]: >  I do not want to re-define all existing / supported safety flags here

there's no need to redefine them all, only the ones needed to change the behaviour

**Comment by Lucas-C** [comment]
No hurry, but I won't merge this without tests, as mentioned in my comment above ;)

**Comment by graingert** [comment]
> No hurry, but I won't merge this without tests, as mentioned in my comment above ;)

yeah of course, I'll add tests once you're happy with the design of the cli

**Comment by graingert** [comment]
@Lucas-C tests added

**Comment by Lucas-C** [comment]
Thanks !
Released in v1.1.3

**Pull request merged**